src/app/components/Application/Main.jsx

```diff
@@ -48,15 +66,30 @@ class Main extends React.Component {
       // Get the array of selected tags found in the book item
       const inSelectedFilter = utils.getSelectedTags(tagArray, selectedFilters);
 
-      if (inSelectedFilter.length &&
-          (inSelectedFilter.length === selectedFilters.length)) {
-        return book;
-      }
-
-      return undefined;
+      return (inSelectedFilter.length && (inSelectedFilter.length === selectedFilters.length));
```

Comment:
The filter function is better used if you return a boolean. If the boolean is true, the data object gets passed, otherwise it doesn't.